convex/lib/honoWithConvex.ts

```diff
@@ -26,35 +67,65 @@ export class HttpRouterWithHono extends HttpRouter {
   }
 
-  // Used to populate the list of routes on the Functions page of the dashboard
+  /**
+   * Returns a list of routed HTTP endpoints.
+   *
+   * These are used to populate the list of routes shown in the Functions page of the Convex dashboard.
+   *
+   * @returns - an array of [path, method, endpoint] tuples.
+   */
   getRoutes = () => {
-    const convexRoutes = this._app.routes.map((route) => {
-      return [route.path, route.method as RoutableMethod] as const;
-    });
-    const result = [];
-    const seen = new Set();
+    const convexRoutes: [string, RoutableMethod, (...args: any) => any][] = [];
+
     // Likely a better way to do this, but hono will have multiple handlers with the same
     // name (i.e. for middleware), so de-duplicate so we don't show multiple routes in the dashboard.
-    // Also source mapping does not play well with this.
-    for (const route of convexRoutes) {
-      const name = `${route[1]} ${route[0]}`;
-      if (seen.has(name)) {
-        continue;
+    const seen = new Set();
+    this._app.routes.forEach((route) => {
+      // Hono uses "ALL" in its router, which is not supported by the Convex router.
+      // Expand this into a route for every routable method supported by Convex.
+      if (route.method === "ALL") {
+        for (const method of ROUTABLE_HTTP_METHODS) {
+          const name = `${method} ${route.path}`;
+          if (!seen.has(name)) {
+            seen.add(name);
+            convexRoutes.push([route.path, method, route.handler]);
+          }
+        }
       } else {
-        seen.add(name);
-        result.push([route[0], route[1], this._handler] as const);
+        const name = `${route.method} ${route.path}`;
+        if (!seen.has(name)) {
+          seen.add(name);
+          convexRoutes.push([
+            route.path,
+            route.method as RoutableMethod,
+            route.handler,
+          ]);
+        }
       }
-    }
-    return result;
+    });
+    return convexRoutes;
   };
 
-  // Used to attribute a request with a particular endpoint for the purposes of
-  // metrics and logs. The method and path should match one of the routes returned
-  // by `getRoutes`
+  /**
+   * Returns the appropriate HTTP endpoint and its routed request path and method.
+   *
+   * The path and method returned are used for logging and metrics, and should
+   * match up with one of the routes returned by `getRoutes`.
+   *
+   * For example,
+   *
+   * ```js
+   * http.route({ pathPrefix: "/profile/", method: "GET", handler: getProfile});
+   *
+   * http.lookup("/profile/abc", "GET") // returns [getProfile, "GET", "/profile/*"]
+   *```
+   *
+   * @returns - a tuple [PublicHttpEndpoint, method, path] or null.
+   */
   lookup = (path: string, method: RoutableMethod | "HEAD") => {
     const match = this._app.router.match(method, path);
     if (match === null) {
-      return [this._handler, method as RoutableMethod, path] as const;
+      return null;
```

Comment:
I forgot that these could just return `null` when there was no match 🤦‍♀️

Reply:
does that preclude the NotFound handling?

Reply:
Ah good point -- if you have middleware on something like `*`, then we get Hono's 404. If you don't, then we get the Convex one. Seems preferable to just always get the Hono 404 so updated.